core-java-modules/core-java-security-4/src/test/java/com/baeldung/keystorealias/KeystoreCertificateNameAliasUnitTest.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.keystorealias;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+
+import org.junit.jupiter.api.Test;
+
+public class KeystoreCertificateNameAliasUnitTest {
+    private static final String KEYSTORE_FILE = "my-keystore.jks";
+    private static final String KEYSTORE_PWD = "storepw@1";
+    private static final String KEYSTORE_ALIAS = "baeldung";
+
+    private KeyStore readKeyStore() throws Exception {
+        KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+        keystore.load(getClass().getResourceAsStream(KEYSTORE_FILE), KEYSTORE_PWD.toCharArray());
+        return keystore;
+    }
+
+    @Test
+    void whenCheckingAliasAndName_thenMatchIsFound() throws Exception {
+        KeyStore keystore = readKeyStore();
+
+        assertThat(keystore.containsAlias(KEYSTORE_ALIAS)).isTrue();
+
+        X509Certificate x509Certificate = (X509Certificate) keystore.getCertificate(KEYSTORE_ALIAS);
+        String owner = x509Certificate.getSubjectX500Principal().getName();
```

Comment:
maybe instead of **owner** it would be better to keep **name** since you are referring to Name in the method and article.

Reply:
sounds good, I've updated the code